### PR TITLE
Allow to specify custom Guice Modules (WIP)

### PR DESCRIPTION
**Pull request by rouazana**
WIP, see TODO in the code, and also:
TODO: use a maven project to generate testModule.jar (see for example james-server-guice-custom-mailets project)

### CODE REVIEW

File: server/container/guice/guice-common/src/main/java/org/apache/james/GuiceJamesServer.java [hunk 1]
```diff
@@ -48,7 +52,15 @@
     public GuiceJamesServer() {
         this(Modules.combine(
                         new CommonServicesModule(),
-                        new MailetProcessingModule()));
+                        new MailetProcessingModule(),
+                        extensionsModules()));
+    }
+
+    private static Module extensionsModules() {
+        // TODO the extensions-jars folder should be taken from filesystem (ie from configuration, not default class path)
```
Comment (chibenwa) [inline server/container/guice/guice-common/src/main/java/org/apache/james/GuiceJamesServer.java]: This requires some form of two stege bindings.... Cf the work I had done.

**Comment by aduprat** [comment]
Build triggered. sha1 is original commit.

**Comment by aduprat** [comment]
Build started sha1 is original commit.

**Comment by aduprat** [comment]
Build triggered. sha1 is original commit.

**Comment by aduprat** [comment]
Build triggered. sha1 is original commit.

**Comment by openpaas-james** [comment]
Build finished. No test results found.

**Comment by openpaas-james** [comment]
Build started sha1 is original commit.

**Comment by openpaas-james** [comment]
Build finished. No test results found.

**Pull request closed without merging**